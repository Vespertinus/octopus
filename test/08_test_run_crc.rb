#!/usr/bin/ruby1.9.1

$:.push 'test/lib'
require 'standalone_env'

class Env < StandAloneEnv
  def config
    super + <<EOD
run_crc_delay = 0.4

object_space[0].enabled = 1
object_space[0].index[0].type = "HASH"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"
EOD
  end
end

Env.clean.with_server do
  ping
  129.times do |i|
    insert [i, i + 1, i + 2]
  end
  sleep 1.1

  puts `./octopus --cat 00000000000000000002.xlog | sed 's/tm:[^ ]* //' | grep run_crc` + "\n"
end
