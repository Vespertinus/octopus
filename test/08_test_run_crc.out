# box.ping()
:pong

# box.insert([0, 1, 2])
1

# box.insert([1, 2, 3])
1

# box.insert([2, 3, 4])
1

lsn:3 scn:3 t:run_crc 0.0.0.0:0 log:0x5cdc7a8e mod:0x6283ebdb
lsn:5 scn:5 t:run_crc 0.0.0.0:0 log:0x3450f038 mod:0x1108d8e8

